aws/internal/nullable/float.go

```diff
@@ -0,0 +1,140 @@
+package nullable
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+const (
+	TypeNullableFloat = schema.TypeString
+)
+
+// Float represents a nullable floating-point value
+type Float string
+
+// IsNull returns true if the value represents a null value
+func (f Float) IsNull() bool {
+	return f == ""
+}
+
+// Value returns the following information about the variable:
+// * value - Returns the value of the variable, or 0 if it is null or an error
+// * null - Returns true if the variable is null
+// * err - Returns an error from parsing the variable, if there is one
+func (f Float) Value() (float64, bool, error) {
+	if f.IsNull() {
+		return 0, true, nil
+	}
+
+	value, err := strconv.ParseFloat(string(f), 64)
+	if err != nil {
+		return 0, false, err
+	}
+	return value, false, nil
+}
+
+// ValidateTypeStringNullableFloat provides custom error messaging for TypeString floats
+// Some arguments require a floating-point value or unspecified, empty field.
+func ValidateTypeStringNullableFloat(v interface{}, k string) (ws []string, es []error) {
```

Comment:
I'm not sure if this full naming style is better, or the shorter names below. Shorter seems to fit better with Go conventions.

Reply:
+1 to shortening the name, it's a bit verbose at the moment.